src/Features/CSharp/Portable/ChangeSignature/CSharpChangeSignatureService.cs

```diff
@@ -406,17 +406,39 @@ private SyntaxNode GetNodeContainingTargetNode(SyntaxNode matchingNode)
             var reorderedParameters = updatedSignature.UpdatedConfiguration.ToListOfParameters();
 
             var newParameters = new List<T>();
+            int index = 0;
             foreach (var newParam in reorderedParameters)
             {
                 var pos = originalParameters.IndexOf(newParam);
                 var param = list[pos];
+
+                // copy whitespace trivia from original position
+                param = TransferLeadingWhitespaceTrivia(param, list[index]);
+
                 newParameters.Add(param);
+                index++;
             }
 
             var numSeparatorsToSkip = originalParameters.Count - reorderedParameters.Count;
             return SyntaxFactory.SeparatedList(newParameters, GetSeparators(list, numSeparatorsToSkip));
```

Comment:
@heejaechang The `GetSeparators` method here already transfers the separator from old syntax (at the correct position). I think this addresses your scenario, but I'll add a test to confirm.

Reply:
`GetSeparators` doesn't behave the way I thought. It just returns the right number of separators from the beginning of `list`. I've added a few tests and the behavior seems reasonable.